### PR TITLE
Python helper lib

### PR DESCRIPTION
**Pull request by rafrombrc**

### CODE REVIEW

File: lib/ensemble/test_format.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        self.assertTrue(_is_point((1, 2)))
+
+
+class ChartTests(unittest.TestCase):
+    output = '''{"data": [{"x": 256, "y": 43.8}, {"x": 512, "y": 21.2}, {"x": 1024, "y": 31.4}, {"x": 2048, "y": 3.6}], "description": "The amount of memory the graphics card has", "labels": {"x": "Memory", "y": ""}, "section": "gfx", "title": "Memory", "units": {"x": "MB", "y": "%"}}'''
+
+    def test_init_failures(self):
+        # string args aren't strings
+        with self.assertRaises(TypeError):
+            chart = Chart(1, "title", "desc", "section")
```
Comment (robhudson) [inline lib/ensemble/test_format.py]: Since you never use the `chart` variable you can just call `Chart` and not assign it to anything.

Comment (robhudson) [inline lib/ensemble/test_format.py]: Same with the handful following this.

File: lib/ensemble/test_format.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    def test_init_failures(self):
+        # string args aren't strings
+        with self.assertRaises(TypeError):
+            report = Report(1, "title", "desc")
```
Comment (robhudson) [inline lib/ensemble/test_format.py]: Another variable that's never used, and those below.

File: lib/ensemble/test_format.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
+import unittest
+
+from format import _is_dict_w_strings
+from format import _is_point
+from format import Chart
+from format import Report
+
```
Comment (robhudson) [inline lib/ensemble/test_format.py]: 2 blank lines after imports

File: lib/ensemble/test_format.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                       {"x": "count", "y": "%"})
+        chart2.add_points([(1, 73.8), (2, 24.3), (4, 1.4), (8, 0.5)])
+        report.add_chart(chart2)
+        output = report.render_json()
+        self.assertEqual(output, self.output)
+
+
+if __name__ == '__main__':
+    unittest.main()
+
```
Comment (robhudson) [inline lib/ensemble/test_format.py]: no blank line at end of file

**Comment by rafrombrc** [comment]
Thanks for the review... updating based on your suggestions now. As for code to drive this, there isn't any. This is code that's intended to be used as a library by folks who are writing data processing jobs, to make it easier for them to generate the JSON that will work w ensemble.

**Comment by robhudson** [comment]
For some reason I thought this was tied to the FHR and was expecting code using this lib code but writing out the FHR json files. My bad. I understand now.

**Comment by openjck** [comment]
Apologies for the slow response. But this is awesome! I'm really impressed with your Python skills.

If you don't mind, I'd like to merge this into a separate repository, maybe something like *ensemble-transformers*. That'll keep things small and organized and allow us to maintain a separate issue tracker for each effort.

**Comment by openjck** [comment]
Pushed to [ensemble-transformers](https://github.com/mozilla/ensemble-transformers)!

**Comment by openjck** [comment]
@rafrombrc I also added you as an admin to that repo

**Pull request closed without merging**